py-monai: new package

qwertos opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

vivian-hafener committed:
New package: py-monai

vivian-hafener committed:
Fixed linked issues with py-monai

qwertos committed:
[py-monai] removed extra line

qwertos committed:
[py-monai]

- New version 1.3.2
- ran black
- update copyright

qwertos committed:
[py-monai] added license

wdconinc commented on var/spack/repos/builtin/packages/py-monai/package.py:
```diff
@@ -0,0 +1,28 @@
+    version("0.8.1", sha256="e1227e6406cc47c23f6846f617350879ceba353915b948d917bf4308b17ea861")
+    version("0.8.0", sha256="a63df7d5a680d9641c223ea090ff843a7d6f20bdb62095bd44f3b0480a4706ed")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
```suggestion
    depends_on("python@3.6:", type=("build", "run"))
    depends_on("python@3.8:", type=("build", "run"), when="@1.2:")
```
https://github.com/Project-MONAI/MONAI/blob/59a7211070538586369afd4a01eca0a7fe2e742e/setup.cfg#L24

wdconinc commented on var/spack/repos/builtin/packages/py-monai/package.py:
```diff
@@ -0,0 +1,28 @@
+    depends_on("py-torch@1.9:", when="@1.3.2:", type=("build", "run"))
+    depends_on("py-numpy@1.17:", type=("build", "run"))
+    depends_on("py-numpy@1.20:", when="@1.3.2:", type=("build", "run"))
+    depends_on("py-ninja", type="build")
```
Usually the build-only dependencies are listed first, so:
1. python version requirements 
2. build dependencies
3. run dependencies

qwertos committed:
[py-monai]

- moved build only dependencies
- specified newer python requirements for newer versions

wdconinc reviewed (approved):
Thanks for the contribution. Looks good!
